src/frontend/src/flows/addDevice/manage/addDevicePickAlias.ts

```diff
@@ -51,4 +58,30 @@ const init = (): Promise<string | null> =>
       event.preventDefault();
       resolve(deviceAlias.value);
     };
+    const deviceInput = document.getElementById(
+      "deviceAlias"
+    ) as HTMLInputElement;
+
+    deviceInput.addEventListener("invalid", () => {
+      let message = "";
+      if (deviceInput.validity.valueMissing) {
+        message = "Please fill out this field.";
+      } else if (deviceInput.validity.patternMismatch) {
+        if (deviceInput.value.startsWith(" ")) {
+          message = "Name can't start with a space.";
+        } else if (
+          deviceInput.value.endsWith(" ") ||
+          deviceInput.value.endsWith("-")
+        ) {
+          message = "Name can't end with a space or hyphen.";
+        } else {
+          message = "Name can't contain special characters.";
+        }
```

Comment:
how about extracting this into a function? I.e.:

```ts
const foo = (validity: ..., input: string): string => {
    if (...) { ... }
    else if (...) { return "Can't end with a space or hyphen"; }
}

...

    const message = foo(deviceInput.validity, deviceInput.value);
```

it'll clarify the code a bit (those nested ifs are a bit daunting) and you may also be able to use it in `register.rs`. What do you think is best/more clear?